src/dex/generic-rfq/rate-fetcher.ts

```diff
@@ -0,0 +1,387 @@
+import BigNumber from 'bignumber.js';
+import { SwapSide } from 'paraswap-core';
+import { BN_0, BN_1, getBigNumberPow } from '../../bignumber-constants';
+import { IDexHelper } from '../../dex-helper';
+import Fetcher from '../../lib/fetcher/fetcher';
+import { Logger, Address, Token } from '../../types';
+import { OrderInfo } from '../paraswap-limit-orders/types';
+import {
+  BigNumberRate,
+  BigNumberRates,
+  MarketResponse,
+  OrderPriceInfo,
+  PairMap,
+  PriceAndAmount,
+  PriceAndAmountBigNumber,
+  Rates,
+  RatesResponse,
+  RFQConfig,
+  RFQFirmRateResponse,
+  RFQPayload,
+} from './types';
+
+export const FIRM_RATE_TIMEOUT_MS = 500;
```

Comment:
It is quite strict

Reply:
What do you prefer ?

Reply:
Maybe 1 sec. at least? But better to discuss on SU